pkg/gui/controllers/local_commits_controller.go

```diff
@@ -430,6 +441,29 @@ func (self *LocalCommitsController) edit(commit *models.Commit) error {
 	})
 }
 
+func (self *LocalCommitsController) quickStartInteractiveRebase() error {
+	if self.c.Git().Status.WorkingTreeState() != enums.REBASE_MODE_NONE {
+		return self.c.ErrorMsg(self.c.Tr.AlreadyRebasing)
+	}
```

Comment:
I'd prefer this to be a DisabledReason, like we do for amendTo and squashAllFixupsAbove. (We are not super consistent with this, some other commands only check it in the handler, but for new ones it would be good to use DisabledReasons rather more than less.)

More thoughts on this: `Status.WorkingTreeState()` is reasonably cheap, but it still needs to hit the file system, so we may not want to use it for DisabledReasons. Maybe we should switch to `Model.WorkingTreeStateAtLastCommitRefresh` for that, but then we should check `Status.WorkingTreeState()` again in the handler just to be sure. I guess.

Reply:
Fair point. Happy to use `WorkingTreeStateAtLastCommitRefresh` and we don't need to do another check inside the handler: it's actually harmless if we skip the check: it just means that we'll get 'YOU ARE HERE' on the head commit. That's not something I'm worried about.

Reply:
> it's actually harmless if we skip the check: it just means that we'll get 'YOU ARE HERE' on the head commit. That's not something I'm worried about.

No, you'd get a cryptic error from git that you are already in a rebase. That would be pretty unpleasant. You could argue that it's so very unlikely that the model state is out of date that it's not worth worrying about it happening, and that I could agree with.

Also, I was thinking more about the two other places where we now use `Status.WorkingTreeState()` for disabled reasons; I think it would be good to change those, too.

Reply:
Ah sorry, I was thinking about the other edge case where the first commit is a merge (or main branch) commit.

> Also, I was thinking more about the two other places where we now use Status.WorkingTreeState() for disabled reasons; I think it would be good to change those, too.

Agreed